storm-client/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -84,13 +86,21 @@ public RateCounter rateCounter(String metricName, String topologyId,
         return gauge;
     }
 
+    @Deprecated
     public <T> Gauge<T> gauge(String name, Gauge<T> gauge, String topologyId, String componentId, Integer taskId, Integer port) {
```

Comment:
the topologyId and port are now available to the registry.  We can remove them from all the methods here.  I can file a JIRA to do this as a follow on.

Reply:
https://issues.apache.org/jira/browse/STORM-3736